spec/system/admin/enterprise_fees_spec.rb

```diff
@@ -191,28 +207,60 @@
       login_as enterprise_user
     end
 
-    it "creates enterprise fees" do
-      visit edit_admin_enterprise_path(distributor1)
-      within(".side_menu") { click_link 'Enterprise Fees' }
-      click_link "Create One Now"
-
-      select distributor1.name,
-             from: 'sets_enterprise_fee_set_collection_attributes_0_enterprise_id'
-      select 'Packing', from: 'sets_enterprise_fee_set_collection_attributes_0_fee_type'
-      fill_in 'sets_enterprise_fee_set_collection_attributes_0_name', with: 'foo'
-      select 'GST', from: 'sets_enterprise_fee_set_collection_attributes_0_tax_category_id'
-      select 'Flat Percent', from: 'sets_enterprise_fee_set_collection_attributes_0_calculator_type'
-      click_button 'Update'
-
-      expect(flash_message).to eq('Your enterprise fees have been updated.')
-
-      # After saving, we should be redirected to the fees for our chosen enterprise
-      expect(page)
-        .not_to have_select 'sets_enterprise_fee_set_collection_attributes_1_enterprise_id',
-                            selected: 'Second Distributor'
-
-      enterprise_fee = EnterpriseFee.find_by name: 'foo'
-      expect(enterprise_fee.enterprise).to eq(distributor1)
+    context "creating an enterprise fee" do
+      before do
+        visit edit_admin_enterprise_path(distributor1)
+        within(".side_menu") { click_link 'Enterprise Fees' }
+        click_link "Create One Now"
+      end
+
+      shared_examples "shared example" do |tax_category, calculator, flash_message, fee_count|
+        context "setting it up as #{tax_category}, with a #{calculator} calculator" do
+          it "triggers the expected message" do
+            select distributor1.name,
+                   from: 'sets_enterprise_fee_set_collection_attributes_0_enterprise_id'
+            select 'Packing', from: 'sets_enterprise_fee_set_collection_attributes_0_fee_type'
+            fill_in 'sets_enterprise_fee_set_collection_attributes_0_name', with: 'foo'
+            select tax_category, 
+from: 'sets_enterprise_fee_set_collection_attributes_0_tax_category_id'
```

Comment:
Funny indenting here. I can see it's due to the challenge of the very long field name.
(Although it's exacerbated by the deep level of indentation too).

I can see this string is repeated a lot: `sets_enterprise_fee_set_collection_attributes_0`. Perhaps we can set that once in advance, then we can do something simpler like

```suggestion
            select tax_category, from: "#{prefix}_tax_category_id"
```

Reply:
Good one, that makes sense too.